File: C++/179. Largest Number.cpp
class Solution {
public:
    static bool c(string a,string b){
        return a+b>b+a;
    }
    string largestNumber(vector<int>& n) {
        vector<string> a;
        for(int i:n){
            a.push_back(to_string(i));
        }
        sort(a.begin(),a.end(),c);
        if(a[0]=="0")
        return "0";
        string b="";
        for(string s:a){
            b+=s;
        }
        return b;
    }
};

File: JAVA/179. Largest Number.java
class Solution {
        public static boolean compare(String a, String b) {
        return (a + b).compareTo(b + a) > 0;
    }
    public String largestNumber(int[] n) {
     List<String> a = new ArrayList<>();
        for (int i : n) {
            a.add(Integer.toString(i));
        }
        a.sort((x, y) -> compare(x, y) ? -1 : 1);
        if (a.get(0).equals("0")) {
            return "0";
        }
        String b="";
        for (String s : a) {
            b+=s;
        }

        return b;
    }
}